SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -606,15 +603,34 @@ void USpatialNetDriver::OnActorSpawned(AActor* Actor)
 	if (!Actor->GetIsReplicated() ||
 		Actor->GetLocalRole() != ROLE_Authority ||
 		!Actor->GetClass()->HasAnySpatialClassFlags(SPATIALCLASS_SpatialType) ||
-		USpatialStatics::IsActorGroupOwnerForActor(Actor))
+		LoadBalanceStrategy == nullptr ||
+		LoadBalanceStrategy->ShouldHaveAuthority(*Actor))
+	{
+		// We only want to delete actors which are replicated and we somehow gain local authority over, when we shouldn't have authority.
+		return;
+	}
+
+	// OnActorSpawned can be called during Player spawn with an invalid position, which breaks the ShouldHaveAuthority check.
```

Comment:
Is this something we want to follow up on - post an error / add a followup Jira ticket? Fixing it might require an engine PR but sounds like something that might throw up more issues in future.

Reply:
A true fix would likely require an engine PR, however I was thinking - what if we make the check also for if an entity exists. The only valid case for this is during Player connection, so we know there will be another check in CreateEntity. I can test that out this afternoon.